node/network/collator-protocol/src/validator_side.rs

```diff
@@ -47,13 +51,26 @@ const COST_CORRUPTED_MESSAGE: Rep = Rep::CostMinor("Message was corrupt");
 /// Network errors that originated at the remote host should have same cost as timeout.
 const COST_NETWORK_ERROR: Rep = Rep::CostMinor("Some network error");
 const COST_REQUEST_TIMED_OUT: Rep = Rep::CostMinor("A collation request has timed out");
-const COST_REPORT_BAD: Rep = Rep::CostMajor("A collator was reported by another subsystem");
+const COST_REPORT_BAD: Rep = Rep::Malicious("A collator was reported by another subsystem");
 const BENEFIT_NOTIFY_GOOD: Rep = Rep::BenefitMinor("A collator was noted good by another subsystem");
 
+// After this long without activity, peers are disconnected.
+#[cfg(not(test))]
+const ACTIVITY_TIMEOUT: Duration = Duration::from_secs(5);
```

Comment:
Why should a collator send an advertise collation every 5 seconds?

And maybe these values should just be configurable, instead of having them hard coded here?

Reply:
@bkchr the point is that the collator shouldn't. this will keep the collation peer-set mostly empty for the purpose of accepting incoming connections from collators who are ready to collate immediately.

Reply:
https://github.com/paritytech/polkadot/issues/2682

i made it configurable but CLI refactor is its own issue

Reply:
> @bkchr the point is that the collator shouldn't. this will keep the collation peer-set mostly empty for the purpose of accepting incoming connections from collators who are ready to collate immediately.

Fine, but why don't we disconnect a collator directly after we told him its collation result? And after connecting he gets like 1 second to send a collation or otherwise we ban him?

Reply:
Yeah, that sounds better. But I don't believe it should block this PR which will solve the basic version of the issue.